backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java

```diff
@@ -281,6 +298,23 @@ public String completeIt()
 		return IDocument.STATUS_Completed;
 	} // completeIt
 
+	private void checkHUUniqueAttributes(@NonNull final AttributeSetInstanceId attributeSetInstanceId)
```

Comment:
Imo this belongs in the https://github.com/metasfresh/metasfresh/blob/master/backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/attribute/impl/HUUniqueAttributesService.java together with the error message. From there, we can later use it in any document where we need to validate a unique attribute from an ASI.